core/domain/html_validation_service.py

```diff
@@ -831,3 +831,88 @@ def get_filename_with_dimensions(old_filename, exp_id):
     new_filename = regenerate_image_filename_using_dimensions(
         old_filename, height, width)
     return new_filename
+
+
+def add_image_id_in_image_tag(html, image_counter):
+    """Adds image_id-with-value in image tag.
+
+    Args:
+        html: str. HTML of content in which image tag present.
+        image_counter: int. Counter for an image id.
+
+    Returns:
+        (new_content_html, image_counter): Tuple. Tuple contains new html
+            of content and image_id counter.
+    """
+    soup = bs4.BeautifulSoup(html.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        image_counter += 1
+        image_id = 'image_id_' + str(image_counter)
+        image['image_id-with-value'] = escape_html(json.dumps(image_id))
+
+    new_content_html = unicode(soup).replace('<br/>', '<br>')
+
+    return (new_content_html, image_counter)
+
+
+def get_image_info_from_html(html):
+    """Extracts info about each image present in the HTML.
+
+    Args:
+        html: str. the HTML in which the images are present.
+
+    Returns:
+        image_info_list: list. List contaning an image info object.
+    """
+    image_info_list = []
+    soup = bs4.BeautifulSoup(html.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        image_id = json.loads(unescape_html(image['image_id-with-value']))
+        filepath = json.loads(unescape_html(image['filepath-with-value']))
+
+        image_info = {
```

Comment:
Please avoid this state/Image object relate stuff in the html_validation services. (I'm not reviewing this file as this file will change once you'll fix the migration method.)